ui/pages/device_code.tsx

```diff
@@ -45,12 +49,21 @@ const DeviceCode: NextPage = () => {
           } else {
             setErrorMessage("Something went wrong, please try again");
           }
-        },
-      );
+        });
     },
     [challenge],
   );
 
+  if (isSuccess) {
+    return (
+      <PageLayout title="Sign in successful">
+        <p>
+          Your device has been successfully registered. You can close this tab.
+        </p>
+      </PageLayout>
+    );
+  }
```

Comment:
The success needs to be in a new page.

The user will be redirected there after they have successfully logged in and given consent. Look at https://warthogs.atlassian.net/browse/IAM-735?focusedCommentId=456723 for a couple of examples.

Reply:
I am confused. This is rendered as a dedicated page as it was shown in the [design file](https://www.figma.com/file/eFR98XGAliMnPUWeBtWzKI/Login-UI?type=design&node-id=393-43679&mode=design&t=fmb6nbeCeP6zTMqz-4). See also the screen below. Or am I missing something?


![image](https://github.com/canonical/identity-platform-login-ui/assets/1155472/44984ce0-a32e-4bcb-8cbb-7eafdc3d5172)

Reply:
Wait, now I am confused as well. I thought that this page is shown to the user when the user code is accepted (if it is accepted successfully we show this page).

Instead we need to show this to a user after they have given consent (the flow is: accept user_code -> login -> give consent).

So this needs to be on a new page (different file, different URL), without a condition.

Reply:
Oh I see, there is a step between entering the code and getting the success screen. :bulb: 

I'll change it to a dedicated page with dedicated url then, so it can be targetted directly.